### PR TITLE
Adding Cribl to Microsoft Azure Solutions Repo

### PR DESCRIPTION
**Pull request by amiracle**
Change(s):

Cribl Steam Solution added
Added a new folder named "Cribl" in the "Solutions" directory
Created a new Cribl sample files in the Samples Folder
Added Cribl-Logo.svg to "Logos" directory
Reasons for Change(s):
Cribl Stream Solution being added into the Microsoft Azure Sentinel repository.
Version 1.0.0 Added.
Tested and validated Solution file with preview function in Sentinel.
 
@microsoft-github-policy-service agree [company=“Cribl"]
@microsoft-github-policy-service agree

### CODE REVIEW

File: Solutions/Cribl/Data Connectors/Solution_Cribl.json
Comment (v-prasadboke) [inline Solutions/Cribl/Data Connectors/Solution_Cribl.json:1]: Is this a CCP data connector

Comment (amiracle) [reply Solutions/Cribl/Data Connectors/Solution_Cribl.json:1]: This is a Solution.

**Review by amiracle — COMMENTED** [review]
Updated Files to pass the validation.

**Review by amiracle — COMMENTED** [review]
Changes Made:
Corrected Sample files added them into a JSON array
Repaired the mainTemplate.json and createUiDefinition.json to work with validation
Updated Solution_Cribl.json
Added files to 1.0.0.zip file
Updated Parsers and updated their names.
Updated SolutionMetadata.json

**Review by amiracle — COMMENTED** [review]
updated the parser file with the fix to the query field. updated image svg file.

**Review by amiracle — COMMENTED** [review]
Updated Solutions under Data Connectors and SolutionMetadta.

**Comment by v-prasadboke** [comment]
Hello @amiracle, 
- Please add graph queries to the Data connector. 

- Add id property to the parsers 

- Also check on Solution metadata's Domain. 

Refer this Solution for more clarification. 
https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Akamai%20Security%20Events

**Review by amiracle — COMMENTED** [review]
Updates

**Review by amiracle — COMMENTED** [review]
Validated updates to the solution and made changes to be compliant with solution validations.

**Comment by v-prasadboke** [comment]
Hello @amiracle, Please create custom table schema named as CriblInternal_CL and CriblInternal at location 
https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables

File: DataConnectors/Cribl/CriblStream.JSON
Comment (v-prasadboke) [inline DataConnectors/Cribl/CriblStream.JSON:1]: Is this Data connector needed. We already have 1 data connector in solution folder.

File: Solutions/Cribl/SolutionMetadata.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{  
+    "publisherId": "criblinc1673975616879",
+	"offerId": "cribl_stream_vm",
```
Comment (v-prasadboke) [inline Solutions/Cribl/SolutionMetadata.json:3]: Include sentinel keyword in offer id 

image

**Comment by v-prasadboke** [comment]
Hello @amiracle, we are waiting for your reply

File: Solutions/Cribl/SolutionMetadata.json (hunk 1 above)
Comment (amiracle) [reply Solutions/Cribl/SolutionMetadata.json:3]: resolved, added a new application that follows the integration requirements, 'microsoft-sentinel-solution-cribl.'

**Comment by v-prasadboke** [comment]
Hello @amiracle, Can you please work on the above requested changes.

**Comment by amiracle** [comment]
> Hello @amiracle,
> 
> * Please add graph queries to the Data connector.
> * Add id property to the parsers
> * Also check on Solution metadata's Domain.
> 
> Refer this Solution for more clarification. https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Akamai%20Security%20Events

This has been completed. 
- The graph queries have been added to the Data connector. 
- Id's have been added to the parsers.
- The Solution domain of Security - Other has been added.

**Comment by amiracle** [comment]
> Hello @amiracle, Please create custom table schema named as CriblInternal_CL and CriblInternal at location https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables

All of the custom tables have been added to the repo:
- CriblAccess_CL
- CriblAudit_CL
- CriblInternal_CL
- CriblUIAccess_CL

**Comment by amiracle** [comment]
@v-prasadboke - I'm trying to decipher the error messages as to why my code is not passing validation. Can you please explain the error codes so that I can properly address them and get this solution added to the repo?

**Comment by v-prasadboke** [comment]
Hello @amiracle, Can we get on a call for this. 
Please share your time zone and availability so that we can plan for the same. 

You can ping me on teams too if needed - v-prasadboke@microsoft.com

**Comment by v-prasadboke** [comment]
reference for parser : https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Snowflake/Parsers/Snowflake.yaml

V3 tool : C:\GitHub\Azure-Sentinel\Tools\Create-Azure-Sentinel-Solution\V3

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**